source/slang/emit.cpp

```diff
@@ -3443,43 +3451,54 @@ struct EmitVisitor
         for(; operandIndex < operandCount; ++operandIndex )
         {
             if(!first) emit(", ");
-            emitIROperand(ctx, inst->getOperand(operandIndex), mode);
+            emitIROperand(ctx, inst->getOperand(operandIndex), mode, kEOp_General);
             first = false;
         }
         emit(")");
+
+        maybeCloseParens(needClose);
     }
 
     void emitIRCallExpr(
         EmitContext*    ctx,
         IRCall*         inst,
-        IREmitMode      mode)
+        IREmitMode      mode,
+        EOpInfo         outerPrec)
     {
         // We want to detect any call to an intrinsic operation,
         // that we can emit it directly without mangling, etc.
         auto funcValue = inst->getOperand(0);
         if(auto irFunc = asTargetIntrinsic(ctx, funcValue))
         {
-            emitIntrinsicCallExpr(ctx, inst, irFunc, mode);
+            emitIntrinsicCallExpr(ctx, inst, irFunc, mode, outerPrec);
         }
         else
         {
-            emitIROperand(ctx, funcValue, mode);
+            auto prec = kEOp_Postfix;
+            bool needClose = maybeEmitParens(outerPrec, prec);
```

Comment:
Yeah, I thought about it, and it might make some of the messier cases involving compound ops with two possible levels of parens easier, but would also be subtle because a closing `}` in the C++ code here could hide actual I/O taking place.